Client/Frontend/Settings/AppSettingsTableViewController.swift

```diff
@@ -78,6 +78,27 @@ class AppSettingsTableViewController: SettingsTableViewController {
             accountSectionTitle = NSAttributedString(string: Strings.FxAFirefoxAccount)
         }
 
+        let engineSettingChanged: ((String)->((Bool)->Void)) = { engineName in
+            let prefName = "sync.engine." + engineName + ".enabledStateChanged"
+            return { enabled in
+                if let _ = prefs.boolForKey(prefName) { // Switch it back to not-changed
```

Comment:
This is really assuming 'changed', isn't it? What's the harm in simply doing `prefs.setBool(enabled, forKey: prefName)` here?

Reply:
In that case enabledStateChanged will be false if we disable the engine which sounds wrong.